### PR TITLE
Add in documentation for reading errors

### PR DESCRIPTION
**Pull request by acollierr17**
The bot currently provides several error codes and responses when there's some error.

This page in the documentation is to address each type of error in detail.

This page allows the user, staff, and developer to educate themselves and troubleshoot ongoing issues. 

CCing @Skelmis onto this to ensure the semantics of this PR are on track. This PR will go from draft to live once I finish the components and ensure I'm not missing anything.

_Edit: Grammar_

### CODE REVIEW

File: docs/errors.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+| 12   | GENERIC_NOT_FOUND           | [click](#generic-not-found)           |
+
+## Error Information
+
+### Suggestion Message Deleted
+This error is thrown when attempting to manage a suggestion which it's corresponding message has been deleted.
+<ErrorEmbed title="Lorem ipsum" description="Hello world" code={1} />
+
+### Missing Permissions
+This error is thrown when the bot is missing permissions to perform an action.
```
Comment (Skelmis) [inline docs/errors.md]: This is raised when the bot is missing permissions to perform an action **and** knows what permissions are missing. The bot will include the required permissions in the error message.

File: docs/errors.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+### Missing Suggestions Channel
+This error is thrown when the bot is missing a configured suggestions channel.
```
Comment (Skelmis) [inline docs/errors.md]: This error is thrown when the bot is missing a configured suggestions channel and the command you are attempting to run requires one.

Something like that

File: docs/errors.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+### Missing Log Channel
+This error is thrown when the bot is missing a configured suggestions log channel.
```
Comment (Skelmis) [inline docs/errors.md]: See comment above

File: docs/errors.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+### Suggestion Not Found
+This error is thrown when the bot is unable to find a suggestion with the given ID.
+
+### Owner Only
+This error is thrown when the bot is unable to perform an action which is only available to the bot owners.
```
Comment (Skelmis) [inline docs/errors.md]: This error is thrown when you attempt to use an action which is only available to the bot owners.

File: docs/errors.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+### Suggestion Content Too Long
+This error is thrown when the bot is unable to submit a suggestion with a content length greater than the configured maximum.
```
Comment (Skelmis) [inline docs/errors.md]: Add a comment about the current max (1000 inclusive)

File: docs/errors.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+### Invalid Guild Config Choice
+This error is thrown when trying to provide an invalid configuration choice.
```
Comment (Skelmis) [inline docs/errors.md]: This error is thrown when you attempt to view an invalid guild configuration choice.

File: docs/errors.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+### Command On Cooldown
+This error is thrown when the bot is on cooldown for a command.
```
Comment (Skelmis) [inline docs/errors.md]: I mean, at risk of stating the obvious here.
Could we add an example embed and include that the user should wait out the time stated on said embed

File: docs/errors.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+### Generic Forbidden
+This error is thrown as a generic response when the bot is forbidden from performing an action.
```
Comment (Skelmis) [inline docs/errors.md]: This is raised when the bot is missing permissions to perform an action **and** cannot figure out what permissions are missing.
If this error continues to occur, you should contact support for further assistance.

Note for Anthony, this is one of only a few codes which will raise a traceback. Viewing said traceback and taking a guess at the missing permissions based on the method called is the only way at this time to figure this one out sadly

File: docs/errors.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+### Unhandled Error
+This error is thrown when the bot is unable to handle an error.
```
Comment (Skelmis) [inline docs/errors.md]: Add a note to please contact support if this continues to occur

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: docs/errors.md (hunk 7 above)
Comment (acollierr17) [reply docs/errors.md]: If you can provide a screenshot or code snippet of how the embed is displayed, I can create the component from there.

**Comment by Skelmis** [comment]
Maybe it's just mobile, but we also have a table saying what code is what error correct?

**Review by Skelmis — APPROVED** [review]

**Pull request merged**